pkg/proc/eval.go

```diff
@@ -1950,6 +1950,26 @@ func (v *Variable) mapAccess(idx *Variable) (*Variable, error) {
 	return nil, fmt.Errorf("key not found")
 }
 
+// Reslice returns a new array, slice or map that starts at index start and contains
+// up to count children.
+func (v *Variable) Reslice(start, count int, cfg LoadConfig) (newV *Variable, err error) {
+	cfg.MaxArrayValues = count
+	switch v.Kind {
+	case reflect.Array, reflect.Slice:
+		newV, err = v.reslice(int64(start), int64(start+count))
+		if err != nil {
+			return nil, err
+		}
+	case reflect.Map:
+		newV = v
+		newV.Children = nil
+		newV.loaded = false
+		newV.mapSkip = start
+	}
+	newV.loadValue(cfg)
```

Comment:
if v is neither an array, slice or map this will be a nil pointer dereference.

Reply:
Fixed.